inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/snapshot/SnapshotBase.java

```diff
@@ -17,7 +17,33 @@
 
 package org.apache.inlong.agent.plugin.sources.snapshot;
 
+import org.apache.inlong.agent.utils.ThreadUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.OutputStream;
+import java.io.ByteArrayOutputStream;
+import java.nio.file.Files;
+import java.util.Base64;
+
 public interface SnapshotBase {
+    Logger LOGGER = LoggerFactory.getLogger(SnapshotBase.class);
```

Comment:
Is it better to create a new `AbstractSnapshot` to implement `SnapshotBase`, and add `file` initialization and these functions in it? Then, sub-snapshots can directly inherit `AbstractSnapshot`.